src/libraries/Microsoft.Extensions.Http/src/DefaultHttpMessageHandlerBuilder.cs

```diff
@@ -28,7 +32,11 @@ public DefaultHttpMessageHandlerBuilder(IServiceProvider services)
             }
         }
 
-        public override HttpMessageHandler PrimaryHandler { get; set; } = new HttpClientHandler();
+        public override HttpMessageHandler PrimaryHandler
+        {
+            get => _primaryHandler ??= CreatePrimaryHandler();
```

Comment:
I had concerns that there are certain workarounds like the one existing in gRPC, where the user sets the handler explicitly to `null!` (it was also discussed a bit here https://github.com/dotnet/runtime/pull/90272/files#r1293537394)

https://github.com/grpc/grpc-dotnet/blob/854e0fd3d0a7da8507ae0f4bf3adac129fb7e10b/src/Grpc.Net.ClientFactory/GrpcClientServiceExtensions.cs#L349-L354

I'll try to use grep app to check whether there are any other users with similar hacks; and I'll also chat with @JamesNK when he's back to see whether he would be able to change the hack to e.g. ConfigureHttpClientDefaults

Reply:
So you don't need to worry about it anymore: https://github.com/grpc/grpc-dotnet/pull/2445

Reply:
Thanks @JamesNK!